Sources/Swift/Integrations/SessionReplay/SentryOnDemandReplay.swift

```diff
@@ -44,20 +45,46 @@ class SentryOnDemandReplay: NSObject, SentryReplayVideoMaker {
     
     var videoWidth = 200
     var videoHeight = 434
+    var videoScale: Float = 1
     var bitRate = 20_000
     var frameRate = 1
     var cacheMaxSize = UInt.max
     
+    private var actualWidth: Int { Int(Float(videoWidth) * videoScale) }
+    private var actualHeight: Int { Int(Float(videoHeight) * videoScale) }
+        
+    init(outputPath: String, workingQueue: SentryDispatchQueueWrapper, dateProvider: SentryCurrentDateProvider) {
+        self._outputPath = outputPath
+        self.dateProvider = dateProvider
+        self.workingQueue = workingQueue
+    }
+        
+    convenience init(withContentFrom outputPath: String, workingQueue: SentryDispatchQueueWrapper, dateProvider: SentryCurrentDateProvider) {
+        self.init(outputPath: outputPath, workingQueue: workingQueue, dateProvider: dateProvider)
+        guard let content = try? FileManager.default.contentsOfDirectory(atPath: outputPath) else { return }
+        
+        _frames = content.compactMap {
+            guard let extensionIndex = $0.lastIndex(of: "."), $0[extensionIndex...] == ".png"
+            else { return SentryReplayFrame?.none }
+            guard let time = Double($0[..<extensionIndex]) else { return nil }
```

Comment:
There's almost always a better way than using ranges to subscript.
```suggestion
            guard $0.hasSuffix(".png") else { return SentryReplayFrame?.none }
            guard let time = Double($0.dropLast(4)) else { return nil }
```